src/main/java/sg/ncl/DataController.java

```diff
@@ -279,7 +295,12 @@ public String getResources(Model model, @PathVariable String datasetId, HttpSess
             String body = responseEntity.getBody().toString();
 
             if (RestUtil.isError(responseEntity.getStatusCode())) {
-                throw new Exception();
+                MyErrorResource error = objectMapper.readValue(body, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+
+                return getStringResponseEntity(exceptionState);
+            } else if (body.equals("All finished.")) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move the "All finished." string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1132)